### DISCUSSION TITLE
City activation and civilians' presence

### DISCUSSION THREAD

**Discussion by Vsauce07** [Vdauphin/HeartsAndMinds#1429] [Question and answer] [answered]
**Hello!**
I would like to know how I can increase the spawn rate of civilians. I currently have the "Civilian density" parameter at 100% and the "Enemy density" at 40%. Still, I don't see a lot of civilians (just 2 or 3) and I see a lot of patrols of enemies. I also would like to know how to increase city activation from 800m to 1800m. _Thanks in advance for the answer!_

**Comment by Vdauphin** [comment, marked as answer]
Hello,
> I currently have the "Civilian density" parameter at 100%

There is only one civilian per entrerable house in the range of `_spawningRadius/3`
First we get houses:
https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/city/activate.sqf#L131
Then we populate:
https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/city/activate.sqf#L160
You can notice the number of civilian also depend on the type of city

So the total number of civilian depend one:
- number of entrerable houses
- cachingRadius [`_spawningRadius = _cachingRadius/2`](https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/city/activate.sqf#L57)
- Type of city

I don't know in your case which one limit the spawning but most of the time is the city radius because some map maker pourly define it like in Lythium (https://github.com/Vdauphin/HeartsAndMinds/pull/753). You can see if it is the issue by enable debug and look at `cachingRadius` by defining [the player spawn radius to 0](https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/param.hpp#L410-L414). Then you will need to calculate the house range which is `cachingRadius/(2 * 3)` and if it does not include houses, then you find the issue

> I also would like to know how to increase city activation from 800m to 1800m.

Just replace `3` by `18` here https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/param.hpp#L414 and add `18` after `8` and `1800m` after `800m`

Cheers

**Reply by Vsauce07** [reply]
Thank you for the answer!
Is there any way I can fix the problem about the civilians' spawn? I was playing on Lythium and Takistan so the problem is probably what you said

**Reply by Vdauphin** [reply]
> Is there any way I can fix the problem about the civilians' spawn?

- For Takistan I would change https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/city/activate.sqf#L131 to `_spawningRadius/2` or even `_spawningRadius` because takistan have low density of houses
- For Lythium, just don't use it
   - If you really want I think you also need to increase the caching radius by `100` of more like that: [`getNumber(_current >> "RadiusA") + getNumber(_current >> "RadiusB") + 100`](https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/city/init.sqf#L61)
   - And add more city manualy https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/define_mod.sqf#L1-L7
